### ISSUE TITLE
Position Encoding Issue!

### ISSUE THREAD

**Issue by kpant14** [jacobjj/motion_planning_transformer#3]
I have been using MPT for one of my projects. I have noticed that the positional encoding used in this implementation is significantly different than the standard transformer model. I could not understand the reason behind using the randomized initial index at each instance. If I change the positional encoding to the 1D standard transformer positional encoding, the model overfits and doesn't learn. Could you explain the rational behind using this scheme. I could not find any reference of it in the paper.

**Comment by jacobjj** [comment]
Training the model by randomly shifting position encoding was instrumental in improving the accuracy of the planner
for maps of larger sizes. The MPT model trained on the position encoding proposed by the OG Transformer paper showed signs of overfitting. This model fails to highlight the region near the boundary of the training map size. 

![large_map_overfit (1)](https://user-images.githubusercontent.com/12293995/174398103-07e7b08d-568e-40a1-a2dc-164799ac8767.png)

One way to resolve this overfitting is to train on maps of various sizes to ensure that the network observes different position encoding. However, the data collection is often computationally expensive, especially for larger maps. What we did was translate the planning space by randomly offsetting the position encoding values, and that is what the code is doing. This is only done during training. 

![random_translation](https://user-images.githubusercontent.com/12293995/174398047-e6c2b57c-60c1-40e6-a521-e51f274df69b.png)

We are working on a new version of the paper with this included, meanwhile hope this helps.
